### PR TITLE
Music: allow AI

### PR DESCRIPTION
**Pull request by breville**
With this change, a level may include the `play AI drums` block by specifying the following field:

```
"toolbox": {
  "includeAi": true
}
```

The `?play-pattern-ai-block=true` URL parameter continues to work as well.

### CODE REVIEW

**Review by mikeharv — COMMENTED** [review]
Am I correct in understanding that this has to be added to the Toolbox JSON field manually?

Once @sanchitmalhotra126's work on the toolbox editor is merged to staging, would we also be able to add a "Include AI" checkbox, similar to the way functions and variables can be added? https://github.com/code-dot-org/code-dot-org/pull/60995

image

If we want to add this to the GUI editor, should that happen in this branch (ie. wait for Sanchit to merge), or should it be a fast follow-up?  

Worth noting, once that branch merges, levelbuilders will be unable to modify the toolbox JSON directly. However, Sanchit also has a branch that will allow editing the level data JSON field. https://github.com/code-dot-org/code-dot-org/pull/61060

I guess my main point is that, depending on the order in which things merge, it's _possible_ that we could get into a state where this new property is supported but can't actually be added.

**Comment by breville** [comment]
> I guess my main point is that, depending on the order in which things merge, it's possible that we could get into a state where this new property is supported but can't actually be added.

Yes, good call.  We just need this setting in today to unblock some internal authoring, so worst case we can edit the level file manually in that situation.

**Comment by mikeharv** [comment]
> > I guess my main point is that, depending on the order in which things merge, it's possible that we could get into a state where this new property is supported but can't actually be added.
> 
> Yes, good call. We just need this setting in today to unblock some internal authoring, so worst case we can edit the level file manually in that situation.

Looks like we merged the JSON editor today, so my concern is much less urgent. Once the AI config option is supported, it can be added that way. We should probably still plan to add it to the GUI later as QOL improvement.

**Review by mikeharv — APPROVED** [review]

**Review by sanchitmalhotra126 — APPROVED** [review]
LGTM! 

For now this seems fine to unblock curriculum and start writing levels with the AI block. I do have some thoughts on how we could store this value more generically (maybe an `experiments: string[]` array?) so that it works with the toolbox editor as well.

**Comment by breville** [comment]
> For now this seems fine to unblock curriculum and start writing levels with the AI block. I do have some thoughts on how we could store this value more generically (maybe an `experiments: string[]` array?) so that it works with the toolbox editor as well.

That sounds good.  It seems like we might more HOC-specific variations coming, so they might help inform this design too.

**Pull request merged**
